torch/_higher_order_ops/wrap.py

```diff
@@ -74,6 +74,30 @@ class TagActivationCheckpoint(HigherOrderOperator):
     def __init__(self):
         super().__init__("tag_activation_checkpoint")
 
+    @staticmethod
+    def divide_kwargs(kwargs):
+        """
+        checkpoint fn can have mixed kwargs between checkpointed fn and
+        checkpoint fn itself. For example
+        >> def gn(x, y, z=None):
+        >>     a = torch.matmul(x, y)
+        >>     if z is not None:
+        >>         return torch.matmul(a, z)
+        >>     return a
+        >> def fn(x, y, z):
+        >>     return torch.cos(checkpoint(gn, x, y, use_reentrant=False, z=z))
+        In the above case, z belongs to checkpointed function gn, but
+        use_reentrant belongs to the checkpoint function. This function splits
+        the kwargs into checkpoint_kwargs and gmod_kwargs (or
+        checkpointed_fn_kwargs).
+        We do sorting to ensure same graph from run to run for better
+        debuggability. It is not required for correctness.
+        """
+        checkpoint_keys = ("use_reentrant", "context_fn", "determinism_check", "debug", "preserve_rng_state")
```

Comment:
Can we use the inspect module to check to signature and grab the kwargs? This would make it more robust. We would still need to manually special case preserve_rng_state (but the comment in checkpoint implies we can make it a regular kwarg, maybe file an issue for it so that someone actually gets rid of it)